python_modules/dagster/dagster_tests/declarative_automation_tests/automation_condition_tests/builtins/test_check_condition.py

```diff
@@ -154,52 +185,41 @@ def blocking3(context) -> AssetCheckResult:
     assert result.total_requested == 0
 
     # blocking1 passes, still not all of them
-    defs.get_implicit_global_asset_job_def().get_subset(
-        asset_check_selection={blocking1.check_key}
-    ).execute_in_process(tags={"passed": ""}, instance=instance)
+    _emit_check({blocking1.check_key}, True)
     result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 0
 
     # blocking2 passes, now all have passed
-    defs.get_implicit_global_asset_job_def().get_subset(
-        asset_check_selection={blocking2.check_key}
-    ).execute_in_process(tags={"passed": ""}, instance=instance)
+    _emit_check({blocking2.check_key}, True)
     result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 1
 
     # blocking3 fails, no impact (as it's not on a dep)
-    defs.get_implicit_global_asset_job_def().get_subset(
-        asset_check_selection={blocking3.check_key}
-    ).execute_in_process(instance=instance, raise_on_error=False)
+    _emit_check({blocking3.check_key}, False)
     result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 1
 
     # nonblocking1 fails, no impact (as it's non-blocking)
-    defs.get_implicit_global_asset_job_def().get_subset(
-        asset_check_selection={nonblocking1.check_key}
-    ).execute_in_process(instance=instance, raise_on_error=False)
+    _emit_check({nonblocking1.check_key}, False)
     result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 1
 
     # now A gets rematerialized, blocking checks haven't been executed yet
     instance.report_runless_asset_event(AssetMaterialization("A"))
+
+    time.sleep(1)
```

Comment:
worth a comment explaining that this is due to sqlite specific timestamp granularity issues?